src/test/java/FooterTest.java

```diff
@@ -68,27 +71,27 @@ public void footer_FooterWebsiteInformation_HasTextCorrectlyDisplayed() {
 
     @Test
     public void footer_FooterCopyrightSection_HasTextCorrectlyDisplayed() {
+        Assertions.assertThat(footer.isCopyrightInformationDisplayed()).isTrue();
         Assertions.assertThat(footer.getFooterCopyrightInformationText()).isEqualTo
                 ("Copyright © 2013-present Magento, Inc. All rights reserved.");
-        Assertions.assertThat(footer.isCopyrightInformationDisplayed()).isTrue();
     }
 
-    public void validateFooterCategoriesAndFooterNamesAndLinks(List<WebElement> footerCategoriesList, List<String> expectedNamesList,
+    public void validateFooterCategoriesNamesAndLinks(List<WebElement> footerCategoriesList, List<String> expectedFooterCategoriesNamesList,
                                                       List<String> expectedCategoriesLinks) {
 
         List<String> actualCategoriesNames = new ArrayList<>();
         List<String> actualCategoriesLinks = new ArrayList<>();
 
         for (WebElement category : footerCategoriesList) {
```

Comment:
It is still invalid. As we discussed offline, the selector for footer list should be
`private final By footerList = By.cssSelector(".footer.content ul.footer.links li a");`
and the method for loop should be adjusted to:
```for (WebElement category : footerCategoriesList) {
            Assertions.assertThat(category.isDisplayed()).isTrue();
            actualCategoriesNames.add(category.getText());
            Assertions.assertThat(category.isEnabled()).isTrue();
            actualCategoriesLinks.add(category.getAttribute(Attribute.HREF.name()));
        }```
        
       The reason is that you extract twice the same element "a", that:
       - is duplication
       - affect tests performance.

Reply:
The reason is that you extract twice the same element "a", that:
       - is duplication
       - affect tests performance.